forte/data/data_store.py

```diff
@@ -0,0 +1,239 @@
+from typing import Union, Type, List, Iterator
+
+import uuid
+from forte.data.ontology.core import EntryType
+from forte.data.base_store import BaseStore
+
+
+class DataStore(BaseStore):
+    def __init__(self):
+        """An implementation from the dataframe-like data store object.
+
+        A collection of NLP entries are stored by types in SortedLists.
+        For example, subtypes of annotations, links, and groups are stored in
+        separate SortedLists.
+        An array records all SortedLists in the format of:
+        [..., Document SortedList, Sentence SortedList, ...].
+        This class records the order of the array and assigns type id,
+        which is the index of each sortedlist, to each entry type.
+
+        Each entry list in the SortedList has the format of
+        [entry_type, tid, begin, end, attr_1, attr_2, ..., attr_n]
+        The first four fields are compulsory for every entry type.
+        Each entry type has a fixed field of attributes.
+        E.g. Document SortedList has lists of structure:
+        [entry_type, tid, begin, end, document_class,sentiment,classifications]
+
+        Args:
+            pack_name (Optional[str], optional): A name for this data store.
+        """
+        super().__init__()
+        self._text = ""
+
+        """
+        The _type_attributes is a private dictionay that provides entry types
+        and their corresponding attribues.
+        The keys are all valid ontology types as strings, including all the
+        types defined in ft.onto and ftx.onto.
+        The values are all the valid attributes for this type, also defined in
+        ft.onto and ftx.onto.
+        This structure is supposed to be generated by another function
+        get_type_attributes() and ready to use in DataStore.
+
+        Example:
+        _type_attributes = {
+            "ft.onto.base_ontology.Token": ["pos", "ud_xpos", "lemma", "chunk", "ner", "sense",
+                    "is_root", "ud_features", "ud_misc"],
+            "ft.onto.base_ontology.Document": ["document_class", "sentiment", "classifications"],
+            "ft.onto.base_ontology.Sentence": ["speaker", "part_id", "sentiment", "classification",
+                        "classifications"],
+        }
+        TODO: implement get_type_attributes() (Issue #570)
```

Comment:
don't put todo in docstring, put in comment. Or simply not using todo, but link to issue to here.